xarray/namedarray/utils.py

```diff
@@ -66,3 +66,26 @@ def to_0d_object_array(value: typing.Any) -> np.ndarray:
     result = np.empty((), dtype=object)
     result[()] = value
     return result
+
+
+class ReprObject:
```

Comment:
This one is not considered `np.typing.ArrayLike`, therefore I don't like it. Is it possible to avoid it?

Example error from #8211:
```
 xarray/core/dataarray.py: note: In member "__init__" of class "DataArray":
xarray/core/dataarray.py:414: error: Incompatible default for argument "data" (default has type "ReprObject", argument has type "T_DuckArray | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float | complex | str | bytes | _NestedSequence[bool | int | float | complex | str | bytes]")  [assignment]
```

Reply:
I don't see why the constructor has a default value of "NA"